### PR TITLE
Fix python example list

### PR DESCRIPTION
**Pull request by therve**
This fixes python list examples on models by not generating the example when we have model objects, and adding the missing closing character when we don't.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Review by jirikuncar — APPROVED** [review]

**Comment by therve** [comment]
@taxpon @frol @mbohlool @cbornet  @kenjones-cisco  @tomplus  @Jyhess @arun-nalla @spacether 🙇

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java [hunk 1]
```diff
@@ -1087,7 +1087,7 @@ private String toExampleValueRecursive(String modelName, Schema schema, Object o
         } else if (ModelUtils.isArraySchema(schema)) {
             if (objExample instanceof Iterable) {
                 // If the example is already a list, return it directly instead of wrongly wrap it in another list
-                return fullPrefix + objExample.toString();
+                return fullPrefix + objExample.toString() + closeChars;
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java:1090]: Why are we not seeing a sample change?
Can you add a test sample which uses this code path?

Comment (therve) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java]: I added a test sample, and it made me realize it didn't fix it the way I expected it to be fixed. I changed the code, let me know if it makes sense.

File: modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml [hunk 2]
```diff
@@ -880,6 +880,15 @@ paths:
           application/json:
             schema:
               $ref: '#/components/schemas/AnimalFarm'
+            examples:
+              simple-list:
+                 summary: Simple list example
+                 description: Should not get into code examples
```
Comment (spacether) [inline modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml:886]: How about adding a test in [PythonClientTest.java](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java#L91) to make sure that the example value for this model is set correctly?

Comment (therve) [reply modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml:886]: I added something!

Comment (spacether) [reply modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml:886]: Thanks!

**Comment by spacether** [comment]
@therve could you rebase on master?
Those jaxrs CI errors are odd and should not be happening.

**Pull request merged**